src/modules/manaFiatGateway/sagas.ts

```diff
@@ -44,23 +50,35 @@ export function createManaFiatGatewaysSaga(config: ManaFiatGatewaySagasConfig) {
 
 function* handleOpenFiatGateway(
   config: ManaFiatGatewaySagasConfig,
-  action: OpenManaFiatGatewayAction
+  action: OpenManaFiatGatewayRequestAction
 ) {
   const { network, gateway } = action.payload
   const { transak: transakConfig, moonPay: moonPayConfig } = config
-  const address: string = yield select(getAddress)
 
-  switch (gateway) {
-    case NetworkGatewayType.TRANSAK:
-      const transak = new Transak(transakConfig, address, network)
-      transak.openWidget(network)
-      break
-    case NetworkGatewayType.MOON_PAY:
-      const widgetUrl = new MoonPay(moonPayConfig).widgetUrl(address, network)
-      window.open(widgetUrl, '_blank', 'noopener,noreferrer')
-      break
-    default:
-      break
+  try {
+    const address: string = yield select(getAddress)
+
+    switch (gateway) {
+      case NetworkGatewayType.TRANSAK:
+        const transak = new Transak(transakConfig, address, network)
+        transak.openWidget(network)
+        break
+      case NetworkGatewayType.MOON_PAY:
+        const moonPay = new MoonPay(moonPayConfig)
+        const widgetUrl: string = yield call(
+          moonPay.getWidgetUrl,
+          address,
+          network
+        )
+        window.open(widgetUrl, '_blank', 'noopener,noreferrer')
+        break
+      default:
+        break
+    }
+
+    yield put(openManaFiatGatewaySuccess())
```

Comment:
Would you mind adding saga tests that exercise the put of the `openManaFiatGatewaySuccess` and `openManaFiatGatewayFailure` actions?